cfme/tests/cloud/test_quota.py

```diff
@@ -0,0 +1,296 @@
+                              secret='{password}'.format(password=fauxfactory.gen_alphanumeric(4))),
+        email=fauxfactory.gen_email(),
+        groups=new_group_project,
+        cost_center="Workload",
+        value_assign="Database",
+    )
+    yield user
+    if user.exists:
+        user.delete()
+
+
+# first arg of parametrize is the list of fixtures or parameters,
+# second arg is a list of lists, with each one a test is to be generated
+# sequence is important here
+# indirect is the list where we define which fixtures are to be passed values indirectly.
+@pytest.mark.parametrize(
+    ["set_child_tenant_quota", "custom_prov_data", "extra_msg", "approve"],
+    [
+        [("cpu", 1), {}, "", False],
+        [("storage", 0.001), {}, "", False],
+        [("memory", 2), {}, "", False],
+        [("vm", 1), {"catalog": {"num_vms": "4"}}, "###", True],
+    ],
+    indirect=["set_child_tenant_quota"],
+    ids=["max_cpu", "max_storage", "max_memory", "max_vms"],
+)
+def test_child_tenant_quota_enforce_via_lifecycle_cloud(
+    request,
+    appliance,
+    provider,
+    new_user_child,
+    set_child_tenant_quota,
+    extra_msg,
+    approve,
+    custom_prov_data,
+    prov_data,
+    provisioning,
+):
+    """Test Child Quota in UI
+
+    Polarion:
+        assignee: ghubale
+        casecomponent: cloud
+        caseimportance: high
+        initialEstimate: 1/8h
+        testSteps:
+            1. Create a child tenant
+            2. Assign quota to child tenant
+            3. Provision instance over the assigned child's quota
+            4. Check whether quota is exceeded or not
+    """
+    with new_user_child:
+        recursive_update(prov_data, custom_prov_data)
+        recursive_update(
+            prov_data,
+            {
+                "request": {
+                    "email": fauxfactory.gen_email(),
+                    "first_name": fauxfactory.gen_alphanumeric(),
+                    "last_name": fauxfactory.gen_alphanumeric(),
+                    "manager_name": "{name}".format(name=fauxfactory.gen_alphanumeric()),
+                }
+            },
+        )
+        prov_data.update({"template_name": provisioning["image"]["name"]})
+        request_description = "Provision from [{template}] to [{vm}{msg}]".format(
+            template=provisioning["image"]["name"], vm=prov_data['catalog']['vm_name'],
+            msg=extra_msg)
+        appliance.collections.cloud_instances.create(
+            prov_data['catalog']['vm_name'],
+            provider,
+            prov_data,
+            auto_approve=approve,
+            override=True,
+            request_description=request_description,
+        )
+        provision_request = appliance.collections.requests.instantiate(request_description)
+        provision_request.wait_for_request(method="ui")
+        assert provision_request.row.reason.text == "Quota Exceeded"
+        request.addfinalizer(provision_request.remove_request)
```

Comment:
This needs to be before the assert, otherwise if the assert fails no cleanup of the request will occur.